File: apps/api/internal/messages/message.go
// Package messages implements the one senior-scoped care-circle stream used by MVP chat.
package messages

import (
	"time"

	"github.com/google/uuid"
)

type Message struct {
	ID           uuid.UUID
	SeniorID     uuid.UUID
	SenderUserID uuid.UUID
	SenderName   string
	Content      string
	CreatedAt    time.Time
}

type Page struct {
	Items       []Message
	NextCursor  string
	UnreadCount int
}

File: packages/contracts/src/message.ts
import type { CursorPage } from './pagination';

/** One message in the senior's care-circle stream. */
export interface CareMessage {
  id: string;
  seniorId: string;
  senderUserId: string;
  senderName: string;
  content: string;
  createdAt: string;
  sentByMe: boolean;
}

export interface MessagePage extends CursorPage<CareMessage> {
  unreadCount: number;
}

export interface CreateMessageRequest {
  content: string;
}

export interface MarkMessagesReadRequest {
  throughMessageId: string;
}
